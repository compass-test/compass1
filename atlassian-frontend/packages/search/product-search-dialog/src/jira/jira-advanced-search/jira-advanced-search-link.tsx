import React from 'react';
import { StyledButton } from './jira-advanced-search.styled';

interface Props {
  href: string;
  dataTestId: string;
  children?: React.ReactElement;
  onClick: (
    href: string,
    e: React.MouseEvent<HTMLSpanElement, MouseEvent>,
  ) => void;
}

export const JiraAdvancedSearchLink = ({
  href,
  onClick,
  children,
  dataTestId,
}: Props) => {
  return (
    <StyledButton
      data-test-id={dataTestId}
      spacing="compact"
      href={href}
      onClick={(e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
        e.preventDefault();
        onClick(href, e);
      }}
    >
      {children}
    </StyledButton>
  );
};
