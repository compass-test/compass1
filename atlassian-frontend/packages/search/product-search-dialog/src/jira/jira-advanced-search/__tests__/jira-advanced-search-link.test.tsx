import React, { ComponentProps } from 'react';
import { shallow } from 'enzyme';
import { JiraAdvancedSearchLink } from '../jira-advanced-search-link';

describe('<JiraAdvancedSearchLink />', () => {
  const onClick = jest.fn();
  const commonProps: ComponentProps<typeof JiraAdvancedSearchLink> = {
    href: 'https://some/href',
    dataTestId: 'mock-data-test-id',
    onClick,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('render matches snapshot', () => {
    const wrapper = shallow(<JiraAdvancedSearchLink {...commonProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick handler is called with the right category', () => {
    const wrapper = shallow(<JiraAdvancedSearchLink {...commonProps} />);

    expect(onClick).toBeCalledTimes(0);
    const mockEvent = ({
      preventDefault: jest.fn(),
    } as any) as React.MouseEvent;
    wrapper.simulate('click', mockEvent);

    expect(onClick).toBeCalledTimes(1);
    expect(onClick).toBeCalledWith(commonProps.href, mockEvent);
    expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
  });
});
