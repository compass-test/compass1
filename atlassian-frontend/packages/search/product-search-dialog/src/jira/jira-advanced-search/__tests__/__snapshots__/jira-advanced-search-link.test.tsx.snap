// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JiraAdvancedSearchLink /> render matches snapshot 1`] = `
<Styled(Button)
  data-test-id="mock-data-test-id"
  href="https://some/href"
  onClick={[Function]}
  spacing="compact"
/>
`;
